course_catalog/tasks.py

```diff
@@ -35,16 +36,26 @@ def get_edx_data(force_overwrite=False):
         settings.EDX_API_URL
         and settings.EDX_API_CLIENT_ID
         and settings.EDX_API_CLIENT_SECRET
+        and settings.OCW_LEARNING_COURSE_BUCKET_NAME
+        and settings.OCW_LEARNING_COURSE_ACCESS_KEY
+        and settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY
     ):
         log.warning("Required settings missing for get_edx_data")
         return
     url = settings.EDX_API_URL
     access_token = get_access_token()
 
+    edx_data = {
+        "count": None,
+        "catalog_url": settings.EDX_API_URL,
+        "results": [],
+    }
+
     while url:
         response = requests.get(url, headers={"Authorization": "JWT " + access_token})
         if response.status_code == 200:
             for course_data in response.json()["results"]:
+                edx_data["results"].append(course_data)
                 try:
                     parse_mitx_json_data(course_data, force_overwrite)
                 except:  # pylint: disable=bare-except
```

Comment:
@mbertrand do you know why L50 is a bare `except` instead of it being catching a base exception i.e. `except Exception`? It's changed from the course-catalog repo and with the pylint disable, I assume it's intentional.

Reply:
That is what's typically used for celery tasks in this project (for example most of `search/tasks.py`.

Reply:
Sure. It'll help ensure the service is kept running, though it introduces the risk of catching exceptions like `KeyboardInterrupt` and `SystemExit ` that we may sometimes want to halt a program.

Would it be helpful if more detail was added to the next line `log.exception("Error encountered parsing MITx json")`, perhaps with something from `course_data`?

Reply:
sure, sounds good